sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala

```diff
@@ -179,9 +179,21 @@ class ParquetToSparkSchemaConverter(
     field match {
       case primitiveColumn: PrimitiveColumnIO => convertPrimitiveField(primitiveColumn, targetType)
       case groupColumn: GroupColumnIO if targetType.contains(VariantType) =>
+        if (groupColumn.getChildrenCount != 2 ) {
+          // We may allow more than two children in the future, so consider this unsupported.
+          throw QueryCompilationErrors.
+            parquetTypeUnsupportedYetError("variant with more than two fields")
+        }
+        val valueIdx = (0 until groupColumn.getChildrenCount)
+            .find(groupColumn.getChild(_).getName == "value")
+        val metadataIdx = (0 until groupColumn.getChildrenCount)
+            .find(groupColumn.getChild(_).getName == "metadata")
```

Comment:
do we need to require the ordering? e.g. value first, then metadata.

Reply:
I don't have a strong opinion. In the test, I specifically made a test case that allows either order. I can't think of a strong reason to allow or forbid arbitrary ordering.

Reply:
I was thinking of our storage model of Variant as being a `struct-of-binary`, and since structs are not sensitive to name order, these fields also shouldn't be.